python/ray/dataframe/utils.py

```diff
@@ -12,6 +12,32 @@
 _NAN_BLOCKS = {}
 
 
+class memoize:
+    """A basic memoizer that cache the input and output of the remote function
+
+    Notes:
+        - This will not work if the argument is mutable. Use with caution.
+    """
+
+    def __init__(self, f):
+        self.old_remote_func = f.remote
+        self.cache = {}
+        self.object_table = ray.global_state.object_table
+
+    def remote(self, *args):
+        """Return cached result if the arguments are cached
+        """
+        args = tuple(args)
+
+        if args in self.cache:
+            cached_result = self.cache[args]
```

Comment:
Can we use an LRU system here instead? This would grow indefinitely. You can cap it to something very large (1000 or so).

Reply:
This will be more important if we add it to additional methods.

Reply:
Good point. I added LRUCache with 1000 capacity, implemented using OrderedDict